### DISCUSSION TITLE
Table - setting the height as 100% and not a number

### DISCUSSION THREAD

**Discussion by avifatal** [rsuite/rsuite#2366] [Q&A] [answered]
Hi,
I using the table, I want the it  to fit the parent. i can't use actual height number because my components needs to be adjusted to any height, Its not predefined
`autoHeight` actually works great, but the scrolling disappear.

https://codesandbox.io/s/rsuite-table-scroll-demo-forked-p8yd82?file=/src/index.js

Possible?
Thanks

**Comment by simonguo** [comment, marked as answer]
Hi @avifatal 
The height of the current Table must be a valid pixel value.  You either make the height of the Table match the height of the container.
```tsx
<div style={{ height: 800}}>
   <Table  height={800}  />
</div>
```
Or dynamically get the height of the container and set the value on the Table.

```tsx
const App = () => {
  const [height, setHeight] = useState(500);
  const containerRef = useRef(null);

  useEffect(() => {
    if (containerRef.current) {
      setHeight(containerRef.current.clientHeight);
    }
  }, []);

  return (
    <div ref={containerRef} style={{ height: 800}} >
      <Table height={height} />
    </div>
  );
};
```

**Reply by avifatal** [reply]
Thank you!
